### PR TITLE
feat: [CDS-49773]: fixing to run-all when dependency between terragru…

### PR DESCRIPTION
**Pull request by vlica-harness**
…nt modules

In this PR are fixed 2 behaviours:
1) When terragrunt run-all within a module which has another terragrunt module dependency, it is prompting us answer with 'yes/no' in case we want to affect that module. That's why it was added echo "y" for run-all init and run-all output commands.
Example:
`Module /Users/vasilelica/harness-repos/prerequisites/terragrunt/dev/local-file-resource depends on module /Users/vasilelica/harness-repos/prerequisites/terragrunt/dev/azure_resource_group, which is an external dependency outside of the current working directory. Should Terragrunt run this external dependency? Warning, if you say 'yes', Terragrunt will make changes in /Users/vasilelica/harness-repos/prerequisites/terragrunt/dev/azure_resource_group as well! (y/n)`
2) When terragrunt run-all from outside the concrete module we previously did terragrunt terragrunt-info, there might be no terragrunt.hcl and the command will fail. That's why we don't need to execute this method. As from this command we take workingDirectory which is used to download state file. For run-all we don't support state file. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42218)

### CODE REVIEW

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42218)* status: 3 file(s) to review. 1 of 2 approvals obtained.

File: 960-api-services/src/main/java/io/harness/terragrunt/v2/TerragruntCommandUtils.java [hunk 1]
```diff
@@ -44,14 +44,16 @@
 @OwnedBy(CDP)
 @UtilityClass
 public class TerragruntCommandUtils {
+  public static final String ECHO_YES_WITH_PLACEHOLDER = "echo \"y\" | %s";
```
Comment (akhilpandey1) [inline 960-api-services/src/main/java/io/harness/terragrunt/v2/TerragruntCommandUtils.java:47]: nit: seems re-usable, should we move to some constants class if it exists for terraform/terragrunt?

**Review by akhilpandey1 — COMMENTED** [review]
Looks good, please see the added comment

File: 960-api-services/src/main/java/io/harness/terragrunt/v2/TerragruntCommandUtils.java (hunk 1 above)
Comment (vlica-harness) [reply 960-api-services/src/main/java/io/harness/terragrunt/v2/TerragruntCommandUtils.java:47]: atm it exists only for terragrunt ng.

**Review by akhilpandey1 — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42218)* status: 1. unresolved discussions.

**Review by akhilpandey1 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42218)* status: 1. unresolved discussions.

**Review by akhilpandey1 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42218#-:-NL01_p67vARlSkBZvPm:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**